protostar/protostar_toml/io/protostar_toml_reader.py

```diff
@@ -1,24 +1,23 @@
 from pathlib import Path
-from typing import Any, Dict, Optional
+from typing import Any, Dict, Optional, cast
 
 import flatdict
 import tomli
 
 from protostar.protostar_toml.protostar_toml_exceptions import (
     NoProtostarProjectFoundException,
 )
-from protostar.utils.protostar_directory import VersionManager
 
 
 class ProtostarTOMLReader:
+    FlatSectionName = str
+
     def __init__(
         self,
-        version_manager: VersionManager,
         protostar_toml_path: Optional[Path] = None,
     ):
         self.path = protostar_toml_path or Path() / "protostar.toml"
-        self._version_manager = version_manager
-        self._cache: Optional[Dict[str, Any]] = None
+        self._cache: Optional[Dict[ProtostarTOMLReader.FlatSectionName, Any]] = None
```

Comment:
`_loaded_config` and `_load_if` would be better name. Cache is misleading, may suggest it's more complicated than it is, that it expires etc.

Reply:
`_loaded_config` can be confused with `ProtostarConfigSection`.

> Cache — hardware or software component that stores data so that future requests for that data can be served faster; the data stored in a cache might be the result of an earlier computation or a copy of data stored elsewhere

The name fits the definition of cache. I like this name because I can later use terms `cache hit` and `cache miss`.

> Cache is misleading, may suggest it's more complicated than it is, that it expires etc.

:/

Reply:
Ok leave cache, it's still a private field so doesn't matter that much